sdk/eventhubs/azure_messaging_eventhubs/src/common/recoverable/connection.rs

```diff
@@ -0,0 +1,473 @@
+// Copyright (c) Microsoft Corporation. All Rights reserved
+// Licensed under the MIT license.
+
+use super::{
+    cbs::RecoverableClaimsBasedSecurity, management::RecoverableManagementClient,
+    receiver::RecoverableReceiver, sender::RecoverableSender,
+};
+use crate::{
+    common::{
+        authorizer::Authorizer,
+        user_agent::{get_package_name, get_package_version, get_platform_info, get_user_agent},
+    },
+    models::AmqpValue,
+    producer::DEFAULT_EVENTHUBS_APPLICATION,
+    ErrorKind, EventHubsError, RetryOptions,
+};
+use async_lock::{Mutex as AsyncMutex, OnceCell};
+use azure_core::{credentials::TokenCredential, http::Url, Result, Uuid};
+use azure_core_amqp::{
+    error::{AmqpErrorCondition, AmqpErrorKind},
+    AmqpClaimsBasedSecurity, AmqpConnection, AmqpConnectionApis, AmqpConnectionOptions, AmqpError,
+    AmqpManagement, AmqpReceiver, AmqpReceiverApis, AmqpReceiverOptions, AmqpSender,
+    AmqpSenderApis, AmqpSession, AmqpSessionApis, AmqpSessionOptions, AmqpSource, AmqpSymbol,
+};
+use std::{collections::HashMap, sync::Arc, time::Duration};
+use tracing::{debug, trace, warn};
+
+/// The recoverable connection is responsible for managing the connection to the Event Hubs service.
+/// It also handles authorization and connection recovery.
+///
+/// * Notes
+///
+/// The way a client uses a `RecoverableConnection` is as follows:
+///   1. Create a new instance of the `RecoverableConnection`.
+///   2. Retrieve an interim object from the `RecoverableConnection`. Supported
+///      interim objects are:
+///    - `AmqpManagement`: Used for management operations.
+///    - `AmqpSender`: Used for sending messages to the Event Hubs service.
+///    - `AmqpReceiver`: Used for receiving messages from the Event Hubs service.
+///    - `AmqpClaimsBasedSecurity`: Used for authorization operations (should not be used directly)
```

Comment:
Markdown requires that sub-bullets be indented to the same level as the text. That said, I don't know if the docs.rs render will do the right thing, but I wouldn't change it. These need to be indented 3 spaces (total, so 2 more). You could always try it with `cargo doc --open`, though.